pkg/webui/components/file-input/index.js

```diff
@@ -168,37 +178,47 @@ export default class FileInput extends Component {
       this.props
 
     return (
-      <div className={style.container}>
-        {image && Boolean(value) && (
-          <img
-            className={classnames(style.image, imageClassName)}
-            alt="Current image"
-            src={value}
-            onError={this.handleImageError}
-            ref={this.imageRef}
+      <div>
+        <div className={style.container}>
+          {image && Boolean(value) && (
+            <img
+              className={classnames(style.image, imageClassName)}
+              alt="Current image"
+              src={value}
+              onError={this.handleImageError}
+              ref={this.imageRef}
+            />
+          )}
+          <Button
+            type="button"
+            aria-controls="fileupload"
+            onClick={this.handleChooseClick}
+            disabled={disabled}
+            message={!value ? message : changeMessage}
+            icon="attachment"
+            secondary
           />
-        )}
-        <Button
-          type="button"
-          aria-controls="fileupload"
-          onClick={this.handleChooseClick}
-          disabled={disabled}
-          message={!value ? message : changeMessage}
-          icon="attachment"
-          secondary
-        />
-        <span className={style.status}>{this.statusMessage}</span>
-        <input
-          name={name}
-          id={id}
-          className={style.input}
-          type="file"
-          onChange={this.handleChange}
-          ref={this.fileInputRef}
-          accept={accept}
-          disabled={disabled}
-          tabIndex="-1"
-        />
+          <span className={style.status}>{this.statusMessage}</span>
+          <input
+            name={name}
+            id={id}
+            className={style.input}
+            type="file"
+            onChange={this.handleChange}
+            ref={this.fileInputRef}
+            accept={accept}
+            disabled={disabled}
+            tabIndex="-1"
+          />
+        </div>
+        <div>
+          {this.state.isLarger && (
+            <React.Fragment>
+              <Icon className={style.warningIcon} icon="error" />
+              <Message className={style.warning} content={m.largeFileWarning} />
+            </React.Fragment>
```

Comment:
```suggestion
            <Notification warning content={m.largeFileWarning}  small />
```

Reply:
Also, the structure should rather be

```javascript
<div className={style.container}>
  … // controls
  <div> // Not sure if the wrapping div is even necessary.
    <Notification />
  </div>
</div>
```

Reply:
With the structure you are suggesting the notification doesn't seem to be in the right place. Because the div has a container style with display flex

<img width="724" alt="Screenshot 2021-11-29 at 10 26 50" src="https://user-images.githubusercontent.com/56658938/143851625-c39d50f5-0375-48ce-8411-9ebd59166a29.png">
.

Reply:
If I do it like so:
<div>
    <Notification />
    <div className={style.container}>
    … // controls
    </div>
</div>

it will work like this:

<img width="724" alt="Screenshot 2021-11-29 at 10 45 27" src="https://user-images.githubusercontent.com/56658938/143854128-683f2c78-f6b7-4e41-924b-55011c4bc68a.png">

Reply:
Ok yeah, this is indeed not working then as I suggested. The thing is though that `<div className={style.container} />` should always be the outermost container. So what you should do here is adapt the CSS so that this is retained.